openeo/rest/job.py

```diff
@@ -136,14 +136,84 @@ def get_results(self) -> "JobResults":
         """
         return JobResults(self)
 
-    def logs(self, offset=None) -> List[LogEntry]:
-        """ Retrieve job logs."""
-        # TODO: option to filter on level? Or move filtering functionality to a separate batch job logs class?
+    def logs(
+        self, offset=None, log_level: Optional[Union[int, str]] = None
+    ) -> List[LogEntry]:
+        """Retrieve job logs.
+
+        :param offset: The last identifier (property ``id`` of a LogEntry) the client has received.
+
+            If provided, the back-ends only sends the entries that occurred after the specified identifier.
+            If not provided or empty, start with the first entry.
+
+            Defaults to None.
+
+        :param log_level: Show only messages of this log level and its higher levels.
+
+            You can use either constants from Python's standard module ``logging``
+            or their names (case insensitive).
+
+            For example:
+                ``logging.INFO``, ``"info"`` or ``"INFO"`` can all be used to show the messages
+                for level ``logging.INFO`` and above, i.e. also ``logging.WARNING`` and
+                ``logging.ERROR`` will be included.
+
+            Defaults to ``logging.ERROR`` (also when explicitly passing log_level=None or log_level="").
+
+        :return: A list containing the log entries for the batch job.
+        """
         url = "/jobs/{}/logs".format(self.job_id)
         logs = self.connection.get(url, params={'offset': offset}, expected_status=200).json()["logs"]
-        entries = [LogEntry(log) for log in logs]
+        log_level = BatchJob._normalize_log_level(log_level)
+        entries = [
+            LogEntry(log)
+            for log in logs
+            if BatchJob._string_to_log_level(log["level"]) >= log_level
+        ]
         return VisualList('logs', data=entries)
 
+    @classmethod
+    def _normalize_log_level(cls, log_level: Optional[Union[int, str]]) -> int:
```

Comment:
I'd move this normalization logic to `openeo/api/logs.py`, because logs are not only for batch jobs (secondary services too)

Reply:
Done.